lib/internal/util/inspect.js

```diff
@@ -861,10 +862,29 @@ function formatError(err, constructor, tag, ctx) {
       }
     }
   }
+  // Ignore the error message if it's contained in the stack.
+  let pos = err.message && stack.indexOf(err.message) || -1;
+  if (pos !== -1)
+    pos += err.message.length;
   // Wrap the error in brackets in case it has no stack trace.
-  const stackStart = stack.indexOf('\n    at');
+  const stackStart = stack.indexOf('\n    at', pos);
   if (stackStart === -1) {
     stack = `[${stack}]`;
+  } else if (ctx.colors) {
+    // Highlight userland code and node modules.
+    let newStack = stack.slice(0, stackStart);
+    const lines = stack.slice(stackStart + 1).split('\n');
+    for (const line of lines) {
+      if (/[/\\]node_modules[/\\]/.test(line)) {
+        newStack += `\n${ctx.stylize(line, 'module')}`;
+      // Node.js core files do not begin with an slash.
```

Comment:
```suggestion
      // Node.js core files do not begin with a slash.
```

Reply:
@bnoordhuis good point. It could also be fooled in case the stack itself is manipulated.

I don't think that there's a perfect solution that is absolutely fool proof but AFAIC that's not a big issue.